### ISSUE TITLE
Provide way to filter out credentials in server startup.

### ISSUE THREAD

**Issue by zquestz** [minio/minio#2749]
## Expected behaviour

Expect a flag like --mask-credentials to star out (*) any sensitive credential information in the server start.
## Actual behaviour

Server startup always shows creds.
## Steps to reproduce the behaviour

Start the server.
## Minio version
- All

NOTE: I am willing to put up a PR to address this. Let me know if that is something you are interested in.

**Comment by harshavardhana** [comment]
> Expect a flag like --mask-credentials to star out (*) any sensitive credential information in the server start.

Can you explain further why would this be needed?  - we wish to avoid new CLI options

**Comment by zquestz** [comment]
Within most PaaS systems all logs are aggregated and then sent to services like Elk or Splunk. This gives credentials to users that may not have policies in the system to see those credentials.

When launching minio server in those environments it is preferable to make sure logs do not contain sensitive server information.

**Comment by harshavardhana** [comment]
> Within most PaaS systems all logs are aggregated and then sent to services like Elk or Splunk. This gives credentials to users that may not have policies in the system to see those credentials.
> 
> When launching minio server in those environments it is preferable to make sure logs do not contain sensitive server information.

Isn't it possible to obfuscate this using `minio server <disk...> >/dev/null` ?

**Comment by zquestz** [comment]
@harshavardhana yes I can dev null all the output, but then the user doesn't get docs and other valuable information. Or see any output at all when starting the server... Ideally the user gets some output that the server started, just not the credentials. For now we are masking it with a really horrid start command:

```
sudo minio --quiet -C /opt/apcera/minio/config server --address ":$PORT" /opt/apcera/minio/data | grep --line-buffered -v AccessKey | grep --line-buffered -v SecretKey | grep --line-buffered -v 'mc config host add'
```

**Comment by harshavardhana** [comment]
> @harshavardhana yes I can dev null all the output, but then the user doesn't get docs and other valuable information. Or see any output at all when starting the server... Ideally the user gets some output that the server started, just not the credentials. For now we are masking it with a really horrid start command:
> 
> sudo minio --quiet -C /opt/apcera/minio/config server --address ":$PORT" /opt/apcera/minio/data | grep --line-buffered -v AccessKey | grep --line-buffered -v SecretKey | grep --line-buffered -v 'mc config host add'

I see we can enhance `minio --quiet` to do this.

**Comment by zquestz** [comment]
That works for me. Want me to put up a PR?

**Comment by harshavardhana** [comment]
> That works for me. Want me to put up a PR?

Please go ahead thanks a lot @zquestz

**Comment by zquestz** [comment]
KK, should be able to get to this in the next day or two. Thanks for the feedback. =)

**Comment by harshavardhana** [comment]
Have a fix for this in patch #2731 we are going to do this using MINIO_SECURE_CONSOLE env variable.

**Comment by zquestz** [comment]
Sounds good to me. =)
